test/unit/metadata.test.js

```diff
@@ -0,0 +1,123 @@
+jest.mock('cross-fetch');
+
+beforeEach(() => {
+    // reset the metadata container to ensure the tests don't interfere with each other
+    // but this also means we need to `require` inside the tests
+    /* eslint-disable global-require */
+    jest.resetModules();
+});
+
+test('get without metadata', async () => {
+    const FetchTool = require('../../src/FetchTool.js');
+    const tool = new FetchTool();
```

Comment:
Can these be moved to beforeEach or possibly a `const` at the top of the file?

Reply:
Each test starts with no modules loaded due to the `jest.resetModules()` in `beforeEach`, so they need to `require` everything fresh. The benefit of that is that each test gets its own metadata state (stored in `scratchFetch.js`, which is also pulled in by `FetchTool.js`). The downside is that each test needs to do its own non-global `require` (or `import`) each time.

I'm basically following the pattern shown here: https://jest-bot.github.io/jest/docs/jest-object.html#jestresetmodules

Reply:
Cool. Thanks for the explanation!